core/font.lua

```diff
@@ -106,7 +109,17 @@ local font = {
       end
     end
     if not options.features then options.features = SILE.settings:get("font.features") end
+    if not options.variations then options.variations = SILE.settings:get("font.variations") end
     if not options.hyphenchar then options.hyphenchar = SILE.settings:get("font.hyphenchar") end
+    if options.variations ~= "" and not SILE.features.font_variations then
+      SU.error([[This build of SILE was compiled with font variations support disabled,
+  likely due to not having the subsetter library included in HarfBuzz >= 6.
+  This document specifies font variations which cannot be correctly rendered.
+  Please rebuild SILE with the necessary library support. Alternatively to procede
+  anyway *incorrectly* render this document run:
+      sile -e 'SILE.features.font_variations = true' ....
+  Or modify the document to remove variations options from font commands.]])
+    end
```

Comment:
We already do a similar check in `shapers/harfbuzz.lua:shaper.getFace()` and it also checks for the face index (which can carry the index of a named instance). It also catches an other potential failures of instantiation (like CFF2) even which harfbuzz-subset is enabled.